src/plugin/file/kml/kmlparser.js

```diff
@@ -841,6 +849,57 @@ plugin.file.kml.KMLParser.prototype.createTreeNode_ = function(el, opt_parent) {
 };
 
 
+/**
+ * Examine styles that are unsupported in OpenLayers KML styles.
+ * @param {Node} node Node.
+ * @private
+ */
+plugin.file.kml.KMLParser.prototype.examineStyles_ = function(node) {
+  var n;
+  for (n = node.firstElementChild; n; n = n.nextElementSibling) {
+    if (n.localName == 'BalloonStyle') {
+      var properties = ol.xml.pushParseAndPop({}, plugin.file.kml.BALLOON_PROPERTY_PARSERS, n, []);
+      if (properties) {
+        var id = node.id || node.getAttribute('id');
+        this.otherStyleMap[id] = properties;
+      }
+    }
+  }
+};
+
+
+/**
+ * Read the KML balloon style.
+ * @param {ol.Feature} feature The feature
+ * @private
+ *
+ * @suppress {accessControls} To allow direct access to feature metadata.
+ */
+plugin.file.kml.KMLParser.prototype.readBalloonStyle_ = function(feature) {
+  var styleUrl = /** @type {string} */ (feature.get('styleUrl'));
+  var style = this.findStyle_(decodeURIComponent(styleUrl), false, true);
+  var text = style.text;
+  var bgColor = style.bgColor || '255, 255, 255, 1';
+  var textColor = style.textColor || '0, 0, 0, 1';
+
+  if (style && text) {
+    var pattern = /[$]\[(.*?)\]/g;
+    var regex = new RegExp(pattern);
+
+    text = text.replace(regex, function(match) {
+      var key = match.slice(2, -1);
```

Comment:
The first capture group will be the [second param to the replacer function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter). No need to do this slice.